### PR TITLE
[ocm-upgrade-scheduler] add support to reject specific versions

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-3641

schema: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/24403

as we discover versions which are not working well for us, we need to have a way to define a list of versions we shouldn't upgrade to.

### CODE REVIEW

File: reconcile/utils/ocm.py [hunk 1]
```diff
@@ -25,23 +26,27 @@ class OCM:
...
     :type offline_token: string
     :type init_provision_shards: bool
     :type init_addons: bool
+    :type blocked_version: list
     """
-    def __init__(self, url, access_token_client_id, access_token_url,
+    def __init__(self, name, url, access_token_client_id, access_token_url,
                  offline_token, init_provision_shards=False,
-                 init_addons=False):
+                 init_addons=False, blocked_versions=None):
```
Comment (Piojo) [inline reconcile/utils/ocm.py:47]: Given how this one is being used later on, can we convert it to a set?

Comment (maorfr) [reply reconcile/utils/ocm.py:47]: it is being passed as a list from app-interface

**Review by jmelis — APPROVED** [review]
LGTM!

We did a sync review, and we agreed to:
- add the `set` as suggested by luis
- add an  additional test for multiple blocked versions
- ensure that there is only one upgrade policy as returned by the API

File: reconcile/utils/ocm.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/ocm.py:48]: added handling for a set

**Comment by jmelis** [comment]
For completeness we also reviewed the logic and we agreed that:
- there is no possibility of race condition here.
- there will be an SOP to know how to block versions.

**Pull request merged**